docs/user-guide/how-to.md

```diff
@@ -123,6 +123,54 @@ jobs:
 Read more about [terraform-docs GitHub Action] and its configuration and
 examples.
 
+## Pre-commit Hook
+
+With [`pre-commit`], you can ensure your Terraform module documentation is kept
+up-to-date each time you make a commit.
+
+First, simply create or update a `.pre-commit-hooks.yaml`
+in the root of your Git repo with at least the following content:
+
+```yaml
+repos:
+  - repo: git://github.com/terraform-docs/terraform-docs
```

Comment:
I'd prefer https protocol.

Reply:
something like `https://github.com/terraform-docs/pre-commit-hook` with effectively two files `README` and `.pre-commit-hooks.yml`?

Reply:
Good idea!

Reply:
Cool, so I'm gonna create the repository now. What would you prefer:

- `terraform-docs/pre-commit`
- `terraform-docs/pre-commit-hook`
- anything else?

@tpansino would you be willing to move the `.pre-commit-hooks.yml` file over there and remove it from this PR?

Reply:
I don't understand - why have a separate repo? That just creates more maintainer overhead for you IMO.

I think @tbugfinder was just asking to switch the proto in the example to https, which doesn't require users to have an SSH set up with GitHub to download the hook. I thought that was pretty reasonable, made the change.

Reply:
Hm, indeed default to https.

On the other hand I do see an option for terraform-docs/pre-commit-hooks (3rd option) as I don't need tool source code just for a hook having few lines.

Reply:
I think after this discussion, we can have the approaches mentioned at https://github.com/terraform-docs/terraform-docs/pull/436#discussion_r596352055, and eventually (outside of scope of this PR) update docs to point them out. So folks can effectively choose between `golang`, `docker`, or `system` hook based on their requirements or liking.

Reply:
So I currently have pushed up all three hook options. They should all work as-is in this repo. 

Can I mark this resolved then? It sounds like creating a separate repo for the `system` hook is a future improvement?

Reply:
I think it's best in not mark these as resolved, for folks looking at this PR in the future to see the discussion (although logically they are resolved)